src/coreclr/src/vm/amd64/JitHelpers_Fast.asm

```diff
@@ -52,378 +52,6 @@ endif
-        je      DoneWithLoop
-
-        ; unroll 2
-        mov     rdx, qword ptr [rdx + OFFSETOF__MethodTable__m_pParentMethodTable]
-        cmp     rcx, rdx
-        je      IsInst
-        test    rdx, rdx
-        je      DoneWithLoop
-
-        ; unroll 3
-        mov     rdx, qword ptr [rdx + OFFSETOF__MethodTable__m_pParentMethodTable]
-        cmp     rcx, rdx
-        je      IsInst
-        test    rdx, rdx
-        jne     CheckParent
-
-    align 16
-    DoneWithLoop:
-        ; Set up the args to call JITutil_ChkCastAny. Note that rcx already contains the MethodTable*
-        mov     rdx, rax    ; rdx = Object*
-
-        ; Call out to JITutil_ChkCastAny to handle the proxy case and throw a rich
-        ; InvalidCastException in case of failure.
-        jmp     JITutil_ChkCastAny
-
-    ; if it is a null instance then rax is null
-    ; if they match then rax contains the instance
-    align 16
-    IsInst:
-        REPRET
-LEAF_END JIT_ChkCastClassSpecial, _TEXT
-
-FIX_INDIRECTION macro Reg
-ifdef FEATURE_PREJIT
-        test    Reg, 1
-        jz      @F
-        mov     Reg, [Reg-1]
-    @@:
-endif
-endm
-
-; PERF TODO: consider prefetching the entire interface map into the cache
-
-; For all bizarre castes this quickly fails and falls back onto the JITutil_IsInstanceOfInterface
-; helper, this means that all failure cases take the slow path as well.
-;
-; This can trash r10/r11
-LEAF_ENTRY JIT_IsInstanceOfInterface, _TEXT
-        test    rdx, rdx
-        jz      IsNullInst
-
-        ; get methodtable
-        mov     rax, [rdx]
-        mov     r11w, word ptr [rax + OFFSETOF__MethodTable__m_wNumInterfaces]
-
-        test    r11w, r11w
-        jz      DoBizarre
-
-        ; fetch interface map ptr
-        mov     rax, [rax + OFFSETOF__MethodTable__m_pInterfaceMap]
-
-        ; r11 holds number of interfaces
-        ; rax is pointer to beginning of interface map list
-    align 16
-    Top:
-        ; rax -> InterfaceInfo_t* into the interface map, aligned to 4 entries
-        ; use offsets of SIZEOF__InterfaceInfo_t to get at entry 1, 2, 3 in this
-        ; block. If we make it through the full 4 without a hit we'll move to
-        ; the next block of 4 and try again.
-
-        ; unroll 0
-ifdef FEATURE_PREJIT
-        mov     r10, [rax + OFFSETOF__InterfaceInfo_t__m_pMethodTable]
-        FIX_INDIRECTION r10
-        cmp     rcx, r10
-else
-        cmp     rcx, [rax + OFFSETOF__InterfaceInfo_t__m_pMethodTable]
-endif
-        je      Found
-        ; move to next entry in list
-        dec     r11w
-        jz      DoBizarre
-
-        ; unroll 1
-ifdef FEATURE_PREJIT
-        mov     r10, [rax + SIZEOF__InterfaceInfo_t + OFFSETOF__InterfaceInfo_t__m_pMethodTable]
-        FIX_INDIRECTION r10
-        cmp     rcx, r10
-else
-        cmp     rcx, [rax + SIZEOF__InterfaceInfo_t + OFFSETOF__InterfaceInfo_t__m_pMethodTable]
-endif
-        je      Found
-        ; move to next entry in list
-        dec     r11w
```

Comment:
Here you can see that we check whether there is anything remaining after checking each entry.